src/components/MovieTile/MovieTile.tsx

```diff
@@ -0,0 +1,55 @@
+import './MovieTile.css';
+import MovieTileModel from '../../models/MovieTileModel';
+import DotsIcon from './../../resources/DotsIcon.png';
+import { useState } from 'react';
+
+interface MovieTileProps {
+  movieModel: MovieTileModel;
+  onClickCallback: (movieName: string) => void;
+  onEditClickCallback: (movieName: string) => void;
+  onDeleteClickCallback: (movieName: string) => void;
+}
+
+function MovieTile({ movieModel, onClickCallback, onEditClickCallback, onDeleteClickCallback }: MovieTileProps) {
+    const [isContextMenuOpen, setContextMenuOpen] = useState(false);
+
+    const toggleContextMenu = () => {
+        setContextMenuOpen(!isContextMenuOpen);
+    };
+        
+    const onClick = () => {
+        onClickCallback && onClickCallback(movieModel?.MovieName);
+    };
+
+    const handleEditClick = () => {
+        onEditClickCallback && onEditClickCallback(movieModel?.MovieName);
+    };
+
+    const handleDeleteClick = () => {
+        onDeleteClickCallback && onDeleteClickCallback(movieModel?.MovieName);
+    };
+    return (
+        <div className="div-MovieTile-General">
+            <div onClick={onClick} className="div-MovieTile-Select">
+                <div className="div-MovieTile-imageContainer">
+                    <img src={movieModel?.ImageUrl} alt='Load error'/>
+                </div>
+                <div className="div-NameYear">
+                    <p>{movieModel?.MovieName}</p>
```

Comment:
Let's add some default values for when some fields are missing in the movie input.

For example: 
` <p>{movieModel?.MovieName || 'Film'}</p>`

OR return null if movie is empty